pkg/agentfs/tar.go

```diff
@@ -144,22 +133,20 @@ func UploadTarball(directory string, presignedUrl string, excludeFiles []string)
 		if err != nil {
 			return fmt.Errorf("failed to calculate relative path for %s: %w", path, err)
 		}
-		// Normalize to forward slashes for tar header names and matching
-		relPath = filepath.ToSlash(relPath)
 
 		for _, exclude := range excludeFiles {
 			if exclude == "" || strings.Contains(exclude, "Dockerfile") {
 				continue
 			}
 
 			if info.IsDir() {
-				if strings.HasPrefix(relPath, exclude+"/") || relPath == exclude {
+				if strings.HasPrefix(relPath, exclude+"/") || strings.HasPrefix(relPath, exclude) {
 					logger.Debugw("excluding directory from tarball", "path", path)
 					return filepath.SkipDir
 				}
 			}
 
-			matched, err := pathpkg.Match(exclude, relPath)
+			matched, err := filepath.Match(exclude, relPath)
 			if err != nil {
 				return nil
```

Comment:
We do still need to write UNIX file paths to the tarball header. I made this change [here](https://github.com/livekit/livekit-cli/pull/666/files#diff-9a9462dcb33d39404e034294b66c3892d69cea945872a95d19041f8f6ad34be4R222) but it seems to have been removed.